diff --git a/vs/sdl/src/cdrom/linux/SDL_syscdrom.c b/vs/sdl/src/cdrom/linux/SDL_syscdrom.c
index 3cc02315c..a7017cb96 100644
--- a/vs/sdl/src/cdrom/linux/SDL_syscdrom.c
+++ b/vs/sdl/src/cdrom/linux/SDL_syscdrom.c
@@ -32,7 +32,7 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <unistd.h>
-#ifdef __LINUX__
+#if defined(__LINUX__) || defined(__ANDROID__) || defined(__linux__)
 #ifdef HAVE_LINUX_VERSION_H
 /* linux 2.6.9 workaround */
 #include <linux/version.h>
